### PR TITLE
Add a tox file to automatically run the tests and add Python 3.9

### PR DESCRIPTION
**Pull request by jon-betts**
This adds a `tox.ini` config file to allow automatic running of the tests. This also:

 * Adds tox environments to run formatting and coverage
 * Advertises the existing Python 3.8 support which was being run against Travis
 * Also adds Python 3.9 support

## Testing notes

You can run the tests against all of the versions of Python by running:

 * `pip install tox`
 * `tox`
 
After much task running you should get a summary like this:

```
  py39: commands succeeded
  py38: commands succeeded
  py37: commands succeeded
  py36: commands succeeded
  py35: commands succeeded
  py27: commands succeeded
  pypy: commands succeeded
  pypy3: commands succeeded
  congratulations :)
```
 
You can also run the formatting and coverage through tox with:

 * `tox -e format`
 * `tox- e coverage`
 
I'm not sure how far / if you'd like to transit these functions to `tox` at all, so for the moment they either replicate or call out to the external functions, but it might be a nice idea to swap it around and let `tox` manage all the dependencies and tasks etc.

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1,8 @@
+3.9.2
+3.8.8
+3.7.10
+3.6.13
+3.5.10
+2.7.18
+pypy-5.7.1
+pypy3.7-7.3.3
```
Comment (jon-betts) [inline .python-version:8]: This is a https://github.com/pyenv/pyenv config file which helps running isolated Python installs locally

File: .travis.yml [hunk 2]
```diff
@@ -7,14 +7,16 @@ python:
...
   - "pypy"
   - "pypy3"
 
 install:
   - pip install pip -U
   - pip install -e .
   - pip install pycodestyle
-  - pip install coveralls
+  # Avoid cryptography compilation problems with pypy and 2.7
+  - pip install coveralls 'cryptography<=3.0;python_version=="2.7"'
```
Comment (jon-betts) [inline .travis.yml:19]: This was the highest version I could get this to compile with

**Comment by un33k** [comment]
@jon-betts Thank you for the PR. Please note that support for py 2.7 is going to be removed on next full version coming soon. With that said, I see that this PR is introducing a dependency to py 2.7 in the yaml file.

**Comment by jon-betts** [comment]
Hi @un33k, if you mean the part about cryptography:

> ```pip install coveralls 'cryptography<=3.0;python_version=="2.7"'```

This is an "environment marker" (https://www.python.org/dev/peps/pep-0508/#environment-markers). This is kind of an if statement which says if the `python_version` is 2.7 then apply this requirement.

So for Python 2.7 environments like CPython 2.7 or pypy the effective requirements will be:

 * `coveralls cryptography<=3.0`
 
But for pypy3 or Python 3.x the effective requirements are:

* `coveralls`

This doesn't add a requirement for Python 2.7.

I'm also happy to rework this to remove `pypy` and 2.7 if you are intending to do that anyway.

**Comment by un33k** [comment]
No worries - I'll remove it on next up version or during badging of this PR.

**Comment by un33k** [comment]
@jon-betts your PR has been pushed to pypi under version 5.0.0. Thx

**Pull request merged**